pkg/controller/zerocapacity/zero_controller.go

```diff
@@ -0,0 +1,465 @@
+		return reconcile.Result{}, instance.UpdatePhase(ZeroFailed, err.Error())
+	}
+
+	if phase == ZeroSucceeded {
+		return reconcile.Result{}, instance.UpdatePhase(ZeroSucceeded, "")
+	}
+
+	// Execution has not completed, or it's state is unknown, wait 1 second before retrying
+	return reconcile.Result{
+		Requeue:      true,
+		RequeueAfter: 1 * time.Second,
+	}, nil
+}
+
+func (z *Controller) cleanupResources(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.Background()
+	meta := metav1.ObjectMeta{
+		Namespace: request.Namespace,
+		Name:      request.Name,
+	}
+	var allErrors error
+
+	// Delete the zero-capacity pod so that it leaves the Infinispan cluster
+	if err := z.Delete(ctx, &corev1.Pod{ObjectMeta: meta}); err != nil && !errors.IsNotFound(err) {
+		allErrors = fmt.Errorf("Unable to delete zero-capacity pod: %w", err)
+	}
+
+	// Delete the configmap as it's no longer required
+	if err := z.Delete(ctx, &corev1.ConfigMap{ObjectMeta: meta}); err != nil && !errors.IsNotFound(err) {
+		allErrors = wrapErr(allErrors, fmt.Errorf("Unable to delete configMap: %w", err))
+	}
+	return reconcile.Result{}, nil
+}
+
+func wrapErr(old, new error) error {
+	if old != nil {
+		return fmt.Errorf("%w; %s", old, new.Error())
+	}
+	return new
+}
+
+func (z *Controller) isZeroPodReady(request reconcile.Request, instance Resource) bool {
+	pod := &corev1.Pod{}
+	key := types.NamespacedName{
+		Name:      request.Name,
+		Namespace: request.Namespace,
+	}
+
+	if err := z.Get(context.Background(), key, pod); err != nil {
+		return false
+	}
+
+	for _, c := range pod.Status.Conditions {
+		if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func (z *Controller) configureBackupPod(name string, configMap *corev1.ConfigMap, ispn *v1.Infinispan, zeroSpec *Spec,
+	pod *corev1.Pod, podSpec corev1.PodSpec) {
+
+	// Utilise the provided PodSpec as a base, overriding fields as required
+	pod.Spec = podSpec
+	var identitiesVol corev1.Volume
+	for _, v := range podSpec.Volumes {
+		if v.Name == ispnCtrl.IdentitiesVolumeName {
+			identitiesVol = v
+		}
+	}
+
+	dataVolName := name + "-data"
+	container := &pod.Spec.Containers[0]
+	container.Name = name
+	container.Resources = zeroSpec.Container.AsResourceRequirements()
+	container.VolumeMounts = []corev1.VolumeMount{
+		{
+			Name:      ispnCtrl.ConfigVolumeName,
+			MountPath: consts.ServerConfigRoot,
+		},
+		{
+			Name:      ispnCtrl.IdentitiesVolumeName,
+			MountPath: consts.ServerSecurityRoot,
+		},
+		// Override the statefulset data volume with Ephemeral vol as we're only interested in data related to CR
+		{
+			Name:      dataVolName,
+			MountPath: ispnCtrl.DataMountPath,
+		},
+		// Mount configured volume at /zero path so that any created content is stored independent of server data
+		{
+			Name:      name,
+			MountPath: zeroSpec.Volume.MountPath,
+		},
+	}
+
+	pod.Spec.SecurityContext.FSGroup = pointer.Int64Ptr(185)
```

Comment:
Tried without this code on OCP 4.5 and it seems to be working without this fix. However adding privileged FSGroup requires `privileged` SCC to be added to the Role that's introduce additional complexity